models/articles.js

```diff
@@ -28,14 +28,17 @@ exports.updateArticle = ({article_id}, body) => {
     })
 }
 
-exports.fetchArticles = () => {
+exports.fetchArticles = async (sortOrder) => {    
+    const queryString = makeQuery(sortOrder)
+    
+    if(sortOrder.topic){
+        await checkExists('topics', 'slug', sortOrder.topic)
+        return db
+        .query(queryString, [sortOrder.topic])
+        .then(({rows}) => rows)
+    }
     return db
-    .query(`
-        SELECT articles.article_id, title, topic, articles.author, articles.created_at, articles.votes, COUNT(comment_id) AS comment_count 
-        FROM articles 
-        LEFT OUTER JOIN comments ON articles.article_id = comments.article_id 
-        GROUP BY articles.article_id 
-        ORDER BY articles.created_at DESC;`)
+    .query(queryString)
     .then(({rows}) => rows)
 }
 
```

Comment:
It's worth adding some logic that checks that the sort_by is one of the valid columns... created_at, votes etc.. if it's not we can reject and there's no need to query the db. Also adds a layer of protection from SQL injection